packages/api/src/utils/externalApi.ts

```diff
@@ -170,22 +170,77 @@ const translateChartDocumentToExternalChart = (
   };
 };
 
-export const translateDashboardDocumentToExternalDashboard = (
-  dashboard: IDashboard,
-): {
+export type ExternalDashboard = {
   id: string;
   name: string;
-  charts: z.infer<typeof externalChartSchemaWithId>[];
-  query: string;
-  tags: string[];
-} => {
-  const { _id, name, charts, query, tags } = dashboard;
+  tiles: ExternalChart[];
+  tags?: string[];
+};
 
+export type ExternalDashboardRequest = {
+  name: string;
+  tiles: ExternalChart[];
+  tags?: string[];
+};
+
+export function translateDashboardDocumentToExternalDashboard(
+  dashboard: Pick<IDashboard, '_id' | 'name' | 'tiles' | 'tags'>,
+): ExternalDashboard {
   return {
-    id: _id.toString(),
-    name,
-    charts: charts.map(translateChartDocumentToExternalChart),
-    query,
-    tags,
+    id: dashboard._id.toString(),
+    name: dashboard.name,
+    tiles: dashboard.tiles,
+    tags: dashboard.tags || [],
   };
+}
+
+// Alert related types and transformations
+export type ExternalAlert = {
+  id: string;
+  name: string | null;
+  message: string | null;
+  threshold: number;
+  interval: string;
+  thresholdType: string;
+  source: string;
+  state: string;
+  channel: any;
+  team: string;
+  tileId?: string;
+  dashboard?: string;
+  savedSearch?: string;
+  groupBy?: string;
+  silenced?: any;
+  createdAt: string;
+  updatedAt: string;
 };
+
+export function translateAlertDocumentToExternalAlert(
+  alert: any,
```

Comment:
should we use alert type from common util instead of using any? if alert is `null` or `undefined` it will error at `alert.toJson`.

Reply:
nice catch. Fixed